controller/payload/payload_scheduler.py

```diff
@@ -694,3 +697,99 @@ def add_information_source_statistics_exclusion(
         if idx % 2 == 0
     ]
     general.add_all(exclusions, with_commit=True)
+
+
+def prepare_sample_records_doc_bin(
+    project_id: str, information_source_id: str
+) -> Tuple[str, List[str]]:
+    sample_records = record.get_attribute_calculation_sample_records(project_id)
+
+    sample_records_doc_bin = get_doc_bin_table_to_json(
+        project_id=project_id,
+        missing_columns="",
+        record_ids=[r[0] for r in sample_records],
+    )
+    project_item = project.get(project_id)
+    org_id = str(project_item.organization_id)
+    prefixed_doc_bin = f"{information_source_id}_doc_bin.json"
+    s3.put_object(
+        org_id,
+        project_id + "/" + prefixed_doc_bin,
+        sample_records_doc_bin,
+    )
+
+    return prefixed_doc_bin, sample_records
+
+
+def run_labeling_function_exec_env(
```

Comment:
this feels like a lot of duplicated code.
can we reuse stuff from run_container ? or create some methods that can be used in both?

Reply:
Closing this for now, but we have to change it